### PR TITLE
🎨Create the Website header - #1

### PR DESCRIPTION
**Pull request by darlene9497**
# Description

This pull request aims to implement a website header for improved navigation on the Elewa website. The header is designed to provide a user experience and has the Elewa logo, transparent background, legible black font, and contact button features.

Fixes https://github.com/italanta/elewa-website/issues/1

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Screenshot
Large screen
Screenshot from 2023-08-09 21-00-46
Tablet screen
Screenshot from 2023-08-11 15-00-12
Mobile screen
Screenshot from 2023-08-11 14-59-31

## How Has This Been Tested?
- [x] Test A - Relied on visual evaluations, as depicted in the screenshot above, and successful compilation on the live server as depicted in the screenshot below.
Screenshot from 2023-08-09 21-03-32

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

File: apps/elewa-website/src/i18n.config.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    return new TranslateHttpLoader(http);
+}
+
+export const I18nModuleConfig = TranslateModule.forRoot({
+    loader: {
+        provide: TranslateLoader,
+        useFactory: HttpLoaderFactory,
+        deps: [HttpClient],
+    },
+});
```
Comment (LemmyMwaura) [inline apps/elewa-website/src/i18n.config.ts:16]: No need for this as I had already set up the translations, just sync the latest changes

File: apps/elewa-website/src/main.ts [hunk 2]
```diff
@@ -1,3 +1,5 @@
+/// <reference types="@angular/localize" />
+
```
Comment (LemmyMwaura) [inline apps/elewa-website/src/main.ts:2]: also this, no need.

File: apps/elewa-website/tsconfig.app.json [hunk 3]
```diff
@@ -2,7 +2,9 @@
   "extends": "./tsconfig.json",
   "compilerOptions": {
     "outDir": "../../dist/out-tsc",
-    "types": []
+    "types": [
+      "@angular/localize"
+    ]
```
Comment (LemmyMwaura) [inline apps/elewa-website/tsconfig.app.json:7]: also here

File: apps/elewa-website/tsconfig.server.json [hunk 4]
```diff
@@ -4,7 +4,8 @@
   "compilerOptions": {
     "outDir": "../../out-tsc/server",
     "types": [
-      "node"
+      "node",
+      "@angular/localize"
```
Comment (LemmyMwaura) [inline apps/elewa-website/tsconfig.server.json:8]: also here

File: libs/elements/layout/header/src/lib/components/header/header.component.scss [hunk 5]
```diff
@@ -0,0 +1,134 @@
+.app-header{
+    display: flex;
+    align-items: center;
```
Comment (LemmyMwaura) [inline libs/elements/layout/header/src/lib/components/header/header.component.scss:3]: Good job composing your styles,

File: package.json [hunk 6]
```diff
@@ -20,12 +20,16 @@
     "@angular/compiler": "16.1.7",
     "@angular/core": "16.1.7",
     "@angular/forms": "16.1.7",
+    "@angular/localize": "^16.1.8",
     "@angular/platform-browser": "16.1.7",
     "@angular/platform-browser-dynamic": "16.1.7",
     "@angular/platform-server": "16.1.7",
     "@angular/router": "16.1.7",
+    "@fortawesome/fontawesome-free": "^6.4.2",
```
Comment (LemmyMwaura) [inline package.json:28]: For font awesome, add a CDN link to the root index.html file

File: libs/elements/layout/header/src/lib/components/header/header.component.html [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+            <li class="nav-item">
+                <a href="#" i18n="@@solutions">
+                    Solutions
+                    <span class="dropdown-indicator">▼</span>
+                </a>
+            </li>
+            <li class="nav-item"><a href="#" i18n="@@projects">Projects</a></li>
+            <li class="nav-item"><a href="#" i18n="@@company">Company</a></li>
+            <li class="nav-item"><a href="#" i18n="@@news">News</a></li>
+        </ul>
```
Comment (LemmyMwaura) [inline libs/elements/layout/header/src/lib/components/header/header.component.html:17]: Write a custom directive to add toggle logic reactivity

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]

**Comment by LemmyMwaura** [comment]
Good job on your PR desc and commits.

**Comment by darlene9497** [comment]
I have made the necessary changes and looking forward to the feedback

File: apps/elewa-website/src/app/toggle-menu.directive.ts [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+@Directive({
+  selector: '[elewaWebsiteToggleMenu]',
+})
+export class ToggleMenuDirective {
+  @HostBinding('class.open') isOpen = false;
+
+  @HostListener('click') toggleOpen() {
+    this.isOpen = !this.isOpen;
+  }
+}
```
Comment (LemmyMwaura) [inline apps/elewa-website/src/app/toggle-menu.directive.ts:12]: The directive should be in the header and not the app module.

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]
Just One final thing. Also, use the directive to toggle the menu from active to inactive. This is not a must you can also do this in the header itself. If you choose option two just delete the directive

**Pull request closed without merging**